### PR TITLE
Fix ray tune layer warning

### PR DESCRIPTION
**Pull request by yinweisu**
*Issue #, if available:*
https://github.com/awslabs/autogluon/issues/2069

*Description of changes:*
This PR addresses point 4 in the above issue. To be notice, I cannot reproduce point 3 now for some reason even though I was able to last week.
* Handle layer as tuple passed to fastai. `nn_torch` doesn't have layers as as nested list so it doesn't have this issue
* Disable tensorboard logging for tabular hpo tune until ray tune handles hyperopt tune issue correctly
* Added `super().__init__()`, which was missing previously

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: core/src/autogluon/core/hpo/executors.py [hunk 1]
```diff
@@ -162,6 +162,7 @@ class RayHpoExecutor(HpoExecutor):
     }
     
     def __init__(self):
+        super().__init__()
```
Comment (Innixma) [inline core/src/autogluon/core/hpo/executors.py:165]: What is the impact of this?

File: core/src/autogluon/core/hpo/executors.py [hunk 2]
```diff
@@ -239,6 +240,9 @@ def execute(
             run,
             RayTuneAdapterFactory
         )
+        # Disable tensorboard logging to avoid layer warning
```
Comment (Innixma) [inline core/src/autogluon/core/hpo/executors.py:243]: Does this speed up training?

File: core/src/autogluon/core/hpo/executors.py [hunk 3]
```diff
@@ -291,6 +296,7 @@ class CustomHpoExecutor(HpoExecutor):
     """Implementation of HpoExecutor Interface, where our custom logic is used as the backend"""
     
     def __init__(self):
+        super().__init__()
```
Comment (Innixma) [inline core/src/autogluon/core/hpo/executors.py:299]: What is the impact of this?

File: tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py [hunk 4]
```diff
@@ -216,6 +216,8 @@ def _fit(self,
         # TODO: calculate max emb concat layer size and use 1st layer as that value and 2nd in between number of classes and the value
         if params.get('layers', None) is not None:
             layers = params['layers']
+            if isinstance(layers, tuple):
+                layers = list(layers)
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py:220]: Is this a bandaid to a bigger problem? Will every model need to keep this in mind and code logic to handle? Is there a way to generically fix this?

File: core/src/autogluon/core/hpo/executors.py (hunk 1 above)
Comment (yinweisu) [reply core/src/autogluon/core/hpo/executors.py:165]: It's more like a bug fix. Previously, forgot to call `super().__init__()` will result in some property not being initialized in some case. It didn't cause problem as I set those property later. But added this just to be safe

File: core/src/autogluon/core/hpo/executors.py (hunk 2 above)
Comment (yinweisu) [reply core/src/autogluon/core/hpo/executors.py:243]: It's likely but I didn't do a benchmark

File: tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py (hunk 4 above)
Comment (yinweisu) [reply tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py:220]: Yes, if a model needs a list as input and is using hyperopt as the search algorithm, this is needed. It's hard to find a way to generically handle it, as we don't know if the input is intended to be a list but got converted to tuple by hyperopt, or it's intended to be a tuple originally.

File: core/src/autogluon/core/hpo/executors.py (hunk 3 above)
Comment (yinweisu) [reply core/src/autogluon/core/hpo/executors.py:299]: ditto

File: tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py (hunk 4 above)
Comment (Innixma) [reply tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py:220]: ah I see, that makes sense regarding the ambiguity.

**Review by Innixma — APPROVED** [review]
LGTM!

**Pull request merged**